### PR TITLE
Async fir for `test/unit_tests/extrinsics/[:-3]`

### PR DESCRIPTION
**Pull request by basfroman**
Async for:
- tests/unit_tests/extrinsics/test_delegation.py
- tests/unit_tests/extrinsics/test_network.py
- tests/unit_tests/extrinsics/test_prometheus.py
- tests/unit_tests/extrinsics/test_registration.py
- tests/unit_tests/extrinsics/test_root.py
- tests/unit_tests/extrinsics/test_senate.py
- tests/unit_tests/extrinsics/test_serving.py

and related files.

### CODE REVIEW

**Review by thealligatorking — APPROVED** [review]

**Review by ibraheem-abe — APPROVED** [review]

File: tests/unit_tests/extrinsics/test_senate.py [hunk 1]
```diff
@@ -34,15 +35,16 @@ def mock_wallet():
         (False, True, False, True, True, True, "happy-path-finalization-true"),
         (True, False, False, True, True, True, "happy-path-inclusion-true"),
         (False, False, False, True, True, True, "happy-path-no_wait"),
-        # Edge cases
+        # # Edge cases
         (True, True, False, True, True, True, "edge-both-waits-true"),
-        # Error cases
+        # # Error cases
```
Comment (thewhaleking) [inline tests/unit_tests/extrinsics/test_senate.py]: I don't think these additional # are necessary

**Review by thewhaleking — APPROVED** [review]
